src/app/pipes/adminfilter.pipe.ts

```diff
@@ -0,0 +1,15 @@
+import { Pipe, PipeTransform } from '@angular/core';
+
+@Pipe({
+  name: 'adminfilter'
+})
+export class AdminfilterPipe implements PipeTransform {
+
+  transform(items: any[], searchText: Object): any {
+    if (!items || !searchText) {
+           return items;
+       }
+     return items.filter(item => item.userName.indexOf(searchText) !== -1);
+  }
```

Comment:
Angular advises against using pipes for searching and sorting ([see documentation here](https://angular.io/guide/pipes#appendix-no-filterpipe-or-orderbypipe)) - can we do this using a service method instead?